### ISSUE TITLE
Error generating thought for cell

### ISSUE THREAD

**Issue by dustinsievers** [getbasedai/creature#6]
Why would I be getting a similar error to this almost all of the time? What does this mean?

Warning: Failed to parse any thoughts from response
  Generated 0 throughts
  [ERR//CRITICAL] >> Error generating thought for cell ####-##-##-####: Failed to generate thought

**Comment by yod22** [comment]
Did you ever find a solution to this?

**Comment by dustinsievers** [comment]
> Did you ever find a solution to this?

@yod22 - No, I haven't, but I'm starting to believe it is because I don't have any OpenRouter credits or payment methods setup yet. But I had thought I read that there would be some free usage at first. But since I didn't see any activity on OpenRouter's site, I'm still not really sure. Once I have some free time to dig in again, I will, but was hoping someone on the development team could provide a pointer.

**Comment by dustinsievers** [comment]
@yod22  - I'm still figuring out the codebase but here is what I did to keep it running despite this issue. In src/models/plan_analysis.rs, at line 47 there is 'let best_plan' etc. I changed it from this:

`let best_plan = plans.iter() .max_by(|a, b| a.score.partial_cmp(&b.score).unwrap());`

Modified line: using unwrap_or to handle None case from partial_cmp
to this:

`let best_plan = plans.iter() .max_by(|a, b| a.score.partial_cmp(&b.score).unwrap_or(std::cmp::Ordering::Equal));`

**Comment by yod22** [comment]
> > Did you ever find a solution to this?
> 
> @yod22 - No, I haven't, but I'm starting to believe it is because I don't have any OpenRouter credits or payment methods setup yet. But I had thought I read that there would be some free usage at first. But since I didn't see any activity on OpenRouter's site, I'm still not really sure. Once I have some free time to dig in again, I will, but was hoping someone on the development team could provide a pointer.

I have credits in my account and had it working but now it's back to not being able to generate thoughts. I'll post any fixes I find

**Comment by dustinsievers** [comment]
@yod22 - Awesome, thanks. I'm really looking forward to fully digesting and modifying this codebase to do something I can benefit from. Much appreciation.

**Comment by yod22** [comment]
Your code worked! I put a few more credits in the account and it's running again.

**Comment by dustinsievers** [comment]
Well, my code will treat any non-comparable scores as equal, allowing max_by to proceed without panicking. partial_cmp(&b.score) returns None if a.score or b.score is a NaN (Not a Number) value. Floating-point NaN values do not have a defined order and thus cannot be compared reliably. If that happens even once, unwrap() will panic.

But I was thinking that this wasn't truly a fix... merely just a way to allow the code to keep running. Because for me, I believe I am still getting these messages: [ERR//CRITICAL] >> Error generating thought for cell ####-##-##-####: Failed to generate thought

Are you saying that after you added credits, you are no longer seeing these messages?

**Comment by mikiad-ai** [comment]
I am also getting this error message. I added credits to my Openrouter Account but it sill wont generate thoughts. Which LLMs are you using through the Openrouter API?

**Comment by yod22** [comment]
I'm using Grok but going to change as it's way too expensive. It seems to stop comms with the creature if your openrouter acct has less then $3 in credits

**Comment by yod22** [comment]
Yes after running your code and adding more creits it's been running without any thought errors. I have noticed it;s repeating itself so not sure if it's actually generating any real thoughts...if that makes sense

**Comment by thibgn** [comment]
Have the same `Failed to generate thought` errors even though I :

- replaced @yod22 code for the `best_plan` (and rebuilt the project)
- have credits on my `OpenRouter` account

I started playing with a creature with $5, it stopped at $1.486 and I have no idea of why
